File: classesCar/Cars/Engine.cpp
#include "Engine.h"

Engine::Engine(char* model, int typeId, int horsePowers, int valueCylindres)
{
    this->id = ++Engine::count;
    strcpy_s(this->model, model);
    this->typeId = typeId;
    this->horsePowers = horsePowers;
    this->valueCylindres = valueCylindres;
}

Engine Engine::InputEngine()
{
    char* model = new char[32];
    int typeId;
    int horsePowers;
    int valueCylindres;

    cout << "Модель двигателя:\t";
    cin >> model;
    do
    {
        cout << "Тип двигателя (Напишите цифру: 1-бензиновый, 2-дизельный):\t";
        cin >> typeId;
    } while (typeId != 1 && typeId != 2);

    cout << "Мощность:\t";
    cin >> horsePowers;
    cout << "Количество цилиндров:\t";
    cin >> valueCylindres;

    Engine engine(model, typeId, horsePowers, valueCylindres);
    return engine;
}

void Engine::PrintEngine()
{
    cout << "Модель двигателя:\t" << this->model << "\n";
    cout << "Тип двигателя:\t";
    if (this->typeId == 1)
    {
        cout <<  "бензиновый\n";
    }
    else
    {
        cout << "дизельный\n";
    }
    cout << "Мощность:\t" << this->horsePowers << "\n";
    cout << "Количество цилиндров:\t" << this->valueCylindres << "\n";
}

int Engine::GetHorsePowers()
{
    return this->horsePowers;
}

File: lz3/classes/Engine.java
package classes;
import java.util.Scanner;


// Двигатель
public class Engine
{
    int id;  // Идентификатор
    char[] model; // Название модели двигателя
    int typeId; // Тип двигателя (1-бензиновый, 2-дизельный)
    int horsePowers; // Мощность в лошадиных силах
    int valueCylindres; // Количество цилиндров

    static int count = 0; // Каждый раз увеличивает id на единицу
    
    public Engine(){ this.id = 0;}
    public Engine(char[] model, int typeId, int horsePowers, int valueCylindres)
    {
        this.model = new char[32];
        this.id = ++count;
        this.model = model;
        this.typeId = typeId;
        this.horsePowers = horsePowers;
        this.valueCylindres = valueCylindres;
    }

    public Engine InputEngine()
    {
        char[] myModel = new char[32];
        int type;

        Scanner in = new Scanner(System.in, "Cp866");  
        System.out.print("Введите название модели двигателя:\t");  
        myModel = in.nextLine().toCharArray(); 
        
        do
        {
            System.out.print("Введите тип двигателя: 1-бензиновый, 2-дизельный:\t");
            type = in.nextInt();
        }while (type != 1 && type != 2);

        System.out.print("Введите мощность в лошадинах силах:\t");
        int powers = in.nextInt();
        
        System.out.print("Введите количество цилиндров:\t");
        int cylindres = in.nextInt();
        // in.close();

        Engine engine = new Engine(myModel, type, powers, cylindres);
        return engine;
    }

    public void PrintEngine()
    {

        System.out.print("\nМодель: ");
        System.out.print(this.model);

        System.out.print("\nТип двигателя: ");
        if(this.typeId == 1)
        {
            System.out.print("бензиновый");
        }
        else
        {
            System.out.print("дизельный");
        }
        System.out.printf("\nМощность: %d", this.horsePowers);

        System.out.printf("\nКоличество цилиндров: %d", this.valueCylindres);
        // System.out.println();

        // in.close();
    }

    public int GetHorsePowers()
    {
        return this.horsePowers;
    }
};

